src/rf/js/src/home/components/modals.js

```diff
@@ -107,9 +188,224 @@ var UploadModal = React.createBackboneClass({
+
+        if(this.state.capture_start && this.state.capture_end) {
+            var start = moment(this.state.capture_start),
+                end = moment(this.state.capture_end);
+            if (end.isBefore(start)) {
+                errors.capture_dates = 'Please enter a capture end that is not before the start.';
+            }
+        }
+
+        if (!this.state.area) {
+            errors.area = 'Please enter an area.';
+        }
+
+        return errors;
+    },
+
+    getServerErrors: function(responseJSON) {
+        var errors = this.getInitialServerErrors();
+
+        if (responseJSON.errors.duplicate_name) {
+            errors.duplicate_name = 'Please choose a name not already in use.';
+        }
+
+        return errors;
+    },
+
+    getUpdatedServerErrors: function(oldState) {
+        // serverErrors should be removed if the input
+        // value that led to the error has been changed.
+        // Of course, the new value could still lead to a server error later.
+        var errors = this.state.serverErrors;
+        if (oldState.name != this.state.name) {
+            errors.duplicate_name = '';
+        }
+        return errors;
+    },
+
+    goToPane: function(paneInd) {
+        this.setState({activePane: paneInd});
+    },
+
+    backPane2: function(e) {
+        e.preventDefault();
+        this.goToPane(1);
+    },
+
+    backPane3: function(e) {
+        e.preventDefault();
+        this.goToPane(2);
+    },
+
+    updateState: function(callback) {
+        // Update the state using values in input elements,
+        // update client and server errors, and then call
+        // optional callback.
+        var self = this,
+            oldState = this.state;
+
+        self.setState(self.getLayerFields(), function() {
+            self.setState({clientErrors: self.getClientErrors()}, function() {
+                self.setState({serverErrors: self.getUpdatedServerErrors(oldState)},
+                              callback || _.noop);
+            });
+        });
+    },
+
+    attemptContinuePane1: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
+            if (self.pane1Valid()) {
+                self.goToPane(2);
+                self.setState({showPane1ClientErrors: false});
+            } else {
+                self.setState({showPane1ClientErrors: true});
+            }
+        });
+    },
+
+    attemptContinuePane2: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
+            if (self.pane2Valid()) {
+                self.goToPane(3);
+                self.setState({showPane2ClientErrors: false});
+            } else {
+                self.setState({showPane2ClientErrors: true});
+            }
+        });
+    },
+
+    attemptSubmit: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
```

Comment:
Consider renaming `updateState`. It isn't immediately clear why you need to update state before saving the layer.